src/main/java/com/zaxxer/nuprocess/internal/BaseEventProcessor.java

```diff
@@ -83,6 +88,8 @@ public void run()
       }
       catch (Exception e) {
          // TODO: how to handle this error?
+         LOGGER.log(Level.WARNING, "Aborting processing loop after unexpected exception (" +
+                 pidToProcessMap.size() + " processes running)", e);
```

Comment:
This logging was instrumental to me in tracking down this issue. If there are other edge cases that cause the `run()` loop to die, this would make them a lot easier to solve going forward.